clients/cloud/python/consumer.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# =============================================================================
+#
+# Consume messages from Confluent Cloud
+# Using Confluent Python Client for Apache Kafka
+#
+# =============================================================================
+
+from confluent_kafka import Consumer, KafkaError
+import sys
+import uuid
+import json
+
+
+if __name__ == '__main__':
+    if len(sys.argv) != 3:
+        sys.stderr.write('Usage: %s <path to Confluent Cloud configuration file> <topic name>\n' % sys.argv[0])
+        sys.exit(1)
+
+    # Read Confluent Cloud configuration for librdkafka clients
+    config_file = sys.argv[1]
+    conf = {}
+    with open(config_file) as fh:
+      for line in fh:
+        line = line.strip()
+        if line[0] != "#" and len(line) != 0:
+          parameter, value = line.strip().split('=', 1)
+          conf[parameter] = value.strip()
+
+    # Create Consumer instance
+    c = Consumer({
+        'bootstrap.servers': conf['bootstrap.servers'],
+        'broker.version.fallback': '0.10.0.0',
```

Comment:
What does this mean?

Reply:
@gwenshap this is how older librdkafka clients negotiated versions with the broker (before you could query the broker for the version) 

@ybyzek I would remove  `broker.version.fallback` and `api.version.fallback.ms=0`. This combination is actually a workaround for a race condition which can cause librdkafka to think the broker is of an older version than it really is. As a result headers could be dropped silently causing much sadness. Either way it's not something people should just be copying and pasting and its unnecessary if you aren't producing with headers.

Reply:
From https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients

> The api.version.request, broker.version.fallback, and api.version.fallback.ms options instruct librdkafka to use the latest protocol version and not fallback to an older version.

Reply:
>  I would remove broker.version.fallback and api.version.fallback.ms=0. 

Don't remove them, they're needed to make sure the latest protocol version is used.
Otherwise the broker will need to perform up/down conversion which slows things down.

Reply:
I don't get it. The version in the cloud is 1.1, right? And we are assuming a new client too (because admin client), so why are we specifying 0.10.0.0 as a version?

If this confuses me (and I have few bits of Kafka experience), at the very least we need to document why this is here.

Reply:
By setting `broker.version.fallback` to a AK version that supports ApiVersionRequest we're forcing the client to always issue ApiVersionRequests, even if the heuristics (connection setup failure, timeout, etc) deems it should downgrade to an older version.
This is for seamlessly supporting AK 0.8 and 0.9 clusters, which I don't think is a sensible default anymore.

I think we'll take the chance to improve these semantics for the v1.0.0 release, in which case neither of these configs will be needed, so I suggest you remove the following configs: broker.version.fallback, api.version.fallback.ms.

Reply:
Ack, removing `broker.version.fallback`, `api.version.fallback.ms`.  Yay for simplification!